src/v1/combo-chart/legend-plugin.ts

```diff
@@ -0,0 +1,125 @@
+import _ from "lodash";
+
+const getOrCreateLegendList = (chart, id) => {
+  const legendContainer = document.getElementById(id);
+  let listContainer = legendContainer.querySelector("ul");
+
+  if (!listContainer) {
+    listContainer = document.createElement("ul");
+    listContainer.style.display = "flex";
+    listContainer.style.flexDirection = "column";
+    listContainer.style.margin = "0";
+    listContainer.style.padding = "0";
+    listContainer.style.height = "95vh";
+    listContainer.style.overflowY = "overlay";
+    listContainer.style.alignItems = "end";
+    listContainer.style.paddingRight = "10px";
+
+    legendContainer.appendChild(listContainer);
+  }
+
+  return listContainer;
+};
+
+export const htmlLegendPlugin = {
+  id: "htmlLegendPlugin",
+  afterUpdate(chart, args, options) {
+    const ul = getOrCreateLegendList(chart, "legend");
+
+    // Remove old legend items
+    while (ul.firstChild) {
+      ul.firstChild.remove();
+    }
+
+    // Reuse the built-in legendItems generator
+
+    const chartMap = {
+      bar: [],
+      line: [],
+      stack: [],
+      scatter: [],
+    };
+    const items = chart.options.plugins.legend.labels.generateLabels(chart);
+
+    items.forEach((style) => {
+      const textParts = style.text.split("- ");
+      const chartType = textParts[1]; // Extracting the chart type (bar, line, etc.)
+
+      switch (chartType) {
+        case "bar":
+          chartMap.bar.push(style);
+          break;
+        case "line":
+          chartMap.line.push(style);
+          break;
+        case "bar-stack":
+          chartMap.stack.push(style);
+          break;
+        case "scatter":
+          chartMap.scatter.push(style);
+          break;
+        default:
+          break;
+      }
+    });
+    _.entries(chartMap).forEach((item) => {
+      const div = document.createElement("div");
+      div.innerHTML = `<p style="padding-bottom: '10px';">${
+        item[0].charAt(0).toUpperCase() + item[0].slice(1)
+      }</p>`;
+      item[1].forEach((item) => {
+        const li = document.createElement("li");
+        li.style.alignItems = "center";
+        li.style.cursor = "pointer";
+        li.style.display = "flex";
+        li.style.flexDirection = "row";
+        li.style.marginLeft = "10px";
```

Comment:
should this not be in percentages?

Reply:
This is for each legend item, I have added a margin.
For legend container I have added in percentage